pennylane/fourier/spectrum.py

```diff
@@ -57,19 +58,34 @@ def stop_at(obj):
                 # parameters - this makes sure no new parameters
                 # were created that depend on the inputs
                 try:
+                    # we need to create a new tape here derived from the class
+                    # of the original tape (i.e., AutogradQuantumTape)
+                    # and queue the ops from the expansion,
+                    # since the get_parameters(trainable_only=True)
+                    # method only distinguishes trainable params
+                    # if the tape is interface-aware
+                    new_tape = tape.__class__()
                     expanded = obj.expand()
+                    for op in expanded.operations:
+                        getattr(new_tape, "_ops").append(op)
```

Comment:
This is pretty hacky, but the tape expansion does it in the same way! Basically we need to make sure that the expansion creates a tape that is interface aware (i.e., not a `QuantumTape` but a `AutogradQuantumTape`)...but the operator does not know of interfaces, only the original tape does...